src/sql/workbench/contrib/notebook/browser/notebookActions.ts

```diff
@@ -425,6 +425,13 @@ export class AttachToDropdown extends SelectBox {
 		if ((connProviderIds && connProviderIds.length === 0) || currentKernel === noKernel) {
 			this.setOptions([msgLocalHost]);
 		} else {
+			// Removes kernel to connection provider mapping of SQL kernel to KUSTO connection
+			for (let kernelAlias of this.model.kernelAliases) {
+				if (connProviderIds.findIndex(e => e === kernelAlias.toUpperCase()) > -1 && currentKernel === 'SQL') {
```

Comment:
Why only for currentKernel? Shouldn't we be resetting the mapping every time we switch kernels regardless of which one it is?